core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2TokenHandler.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.server.security.oauth2;
+
+public interface OAuth2TokenHandler
+{
+    void setAccessToken(String hashedState, String accessToken);
```

Comment:
It's not a hashed state.

Reply:
Actually it is, this is called during the OAuth2 callback which only has the hashedState (hashed UUID) present in the state JWT.

Reply:
That an authId is used under the covers in `OAuth2TokenExchange` is an implementation detail that doesn't apply to the interface which just provides an API.